python/python/lance/ray/sink.py

```diff
@@ -46,22 +46,8 @@ def _pd_to_arrow(
         tbl.schema = tbl.schema.remove_metadata()
         return tbl
     elif isinstance(df, pa.Table):
-        fields = df.schema.names
-        new_columns = []
-        new_fields = []
-        for field in fields:
-            col = df[field]
-            new_field = pa.field(field, col.type)
-            if (
-                pa.types.is_null(col.type)
-                and schema.field_by_name(field).type == pa.string()
-            ):
-                new_field = pa.field(field, pa.string())
-                col = pa.compute.if_else(pa.compute.is_null(col), NONE_ARROW_STR, col)
-            new_columns.append(col)
-            new_fields.append(new_field)
-        new_schema = pa.schema(fields=new_fields)
-        new_table = pa.Table.from_arrays(new_columns, schema=new_schema)
+        new_table = df.to_pydict()
+        new_table = pa.Table.from_pydict(new_table, schema=schema)
```

Comment:
```suggestion
        if schema is None:
            return df
        return df.cast(schema)
```

`to_pydict` and `from_pydict` are expensive.  Does this cast work for you?

Reply:
Yes! Perfect!